src/rsa/padding.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2015-2016 Brian Smith.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY
+// SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+// OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+// CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+use untrusted;
+use digest;
+
+#[cfg(feature = "rsa_signing")]
+use rand;
+
+#[cfg(feature = "rsa_signing")]
+pub trait Encoding {
+    fn encode(&self, msg: &[u8], out: &mut [u8], rng: &rand::SecureRandom)
+              -> Result<(), ()>;
+}
+
+pub trait Verification {
+    fn verify(&self, msg: untrusted::Input, encoded: untrusted::Input) 
```

Comment:
The "encoded" parameter name does not make much sense to me in the context of a `Verification` trait. The concept that `Verification` is implemented for an `Encoding` is also not that intuitive. It seems better to me that the `Encoding`'s single responsibility is to wrap/unwrap, and that verification will be left to the implementer of the `signature::VerificationAlgorithm` trait.